stanza/message.go

```diff
@@ -25,10 +25,29 @@ type Message struct {
 	Type    MessageType `xml:"type,attr,omitempty"`
 }
 
+func (t *MessageType) UnmarshalXMLAttr(attr xml.Attr) error {
+	switch attr.Value {
```

Comment:
Another thought: what if, instead of all the marshaling and unmarshaling, we added a method on `Message`, something like:

```
// TypeAttr returns the normalized message type for the message stanza.
// It will always return one of the constants defined in this package, even if
// the unmarshaled wire value was an undefined value.
// When checking the message type, this should normally be used over the struct
// field.
func (m Message) TypeAttr() MessageType {
	switch m.Type {
	case ChatMessage, ErrorMessage, GroupChatMessage, HeadlineMessage:
		return m.Type
	}
	return NormalMessage
}
```

This would let use get the wire value (`msg.Type`) or the normalized value (`msg.TypeAttr()`) in case we need to differentiate between the two for some reason. I'm not sure if it's a better idea or not though; I'd still like to get your thoughts.

Reply:
I've written up some more comprehensive thoughts in #272; however, I'm now leaning towards what you've already done here.

Reply:
> I've written up some more comprehensive thoughts in #272; however, I'm now leaning towards what you've already done here.

Well, I guess then I don't have to do anything else here.

I think this could have been much cleaner if the types were stronger à la  some kind of enumerated type, and not strings, as RFC doesn't seem to allow other values, and type-system is not enforcing this.

Reply:
All tests are passing now. Is there anything in terms of feedback, which is amiss ?

Thanks!

Reply:
I'll give it another look over soon; thanks!